mscore/resourceManager.cpp

```diff
@@ -264,6 +414,343 @@ bool ResourceManager::verifyLanguageFile(QString filename, QString hash)
       return verifyFile(local, hash);
       }
 
+//---------------------------------------------------------
+//   GitHub utility functions
+//---------------------------------------------------------
+static inline QString githubLatestReleaseAPI(const QString &user, const QString &repo) { return "https://api.github.com/repos/" + user + "/" + repo + "/releases"; }
+static inline QString githubCommitAPI(const QString &user, const QString &repo, const QString &ref = "")
+      {
+      return "https://api.github.com/repos/" + user + "/" + repo + "/commits" + (ref.isEmpty() ? "" : ('/' + ref));
+      }
+static inline QString githubLatestArchiveURL(const QString &user, const QString &repo, const QString &branch = "master")
+      {
+      //return "https://api.github.com/repos/" + user + "/" + repo + "/zip/" + branch;
+      return "https://github.com/" + user + '/' + repo + "/archive/" + branch + ".zip";
+      }
+static QString getLatestCommitSha(const QString& user, const QString &repo, const QString &branch = "master")
+      {
+      QString api_url = githubCommitAPI(user, repo, branch);
+      DownloadUtils json;
+      json.setTarget(api_url);
+      json.download();
+      QJsonDocument json_doc = QJsonDocument::fromJson(json.returnData());
+      QJsonObject latest_commit;
+      if (json_doc.isArray())
+            // there are  multiple commits, fetch the latest
+            latest_commit = json_doc.array().first().toObject();
+      else
+            latest_commit = json_doc.object();
+      if(latest_commit.contains("sha"))
+            return latest_commit["sha"].toString();
+      else return {};
+      }
+static inline bool isCompatibleRelease(const QString& branch)
+      {
+      if (branch == "master" || branch=="3.x") return true;
+      if (branch.contains("2.x") || branch.contains("version2") || branch.contains("musescore2") || branch.contains("musescore 2")) return false;
+      if (branch.contains("3.x") || branch.contains("version3") || branch.contains("musescore3") || branch.contains("musescore 3")) return true;
+      if (branch.contains("2") && !branch.contains("3")) return false;
+      return true;
+      }
+static inline int CompatEstimate(QString branch)
+      {
+      branch = branch.toLower();
+      if (branch.contains("2.x") || branch.contains("version2") || branch.contains("musescore2") || branch.contains("musescore 2")) return -2;
+      if (branch.contains("3.x") || branch.contains("version3") || branch.contains("musescore3") || branch.contains("musescore 3")) return 2;
+      if (branch.contains("2") && !branch.contains("3")) return -1;
+      if (branch.contains("3") && !branch.contains("2")) return 1;
+      return 0;
+      }
+static bool getLatestRelease(QJsonDocument& releases, int& release_id, QString& link) {
+      // By default, releases returned from GitHub are sorted from the latest commit to the oldest
+      foreach(const auto&release, releases.array()) {
```

Comment:
space between & and release, also use for (... : ...)

Reply:
Resolved in 77701599820befd5e5381af2075d63e6b5b3dd85.